lib/full-site-editing/edit-site-page.php

```diff
@@ -76,6 +76,35 @@ function gutenberg_get_editor_styles() {
 	return $styles;
 }
 
+/**
+ * Return the correct template for the site's home page.
+ *
+ * @return array|null A template object, or null if none could be found
+ */
+function gutenberg_edit_site_resolve_home_template() {
+	$show_on_front = get_option( 'show_on_front' );
+	$front_page_id = get_option( 'page_on_front' );
+
+	if ( 'page' === $show_on_front && $front_page_id ) {
+		return array(
+			'postType' => 'page',
+			'postId'   => $front_page_id,
+		);
+	}
+
+	$hierarchy = array( 'front-page', 'home', 'index' );
+	$template  = gutenberg_resolve_template( 'home', $hierarchy, '' );
```

Comment:
Omitting the third parameter here prevents error when a high-level PHP template exists in a hierarchy - https://github.com/WordPress/gutenberg/issues/37236#issuecomment-1039744171.

I don't think resolving to PHP templates is helpful for Site Editor. So instead, we fall back to the closest block template.